### PR TITLE
bluez: 5.75 → 5.76

### PR DESCRIPTION
**Pull request by szaffarano**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
redirect this PR into Staging branch.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Redirect this PR into Staging branch.

**Review by AndersonTorres — APPROVED** [review]

**Comment by K900** [comment]
We probably want to pick https://github.com/bluez/bluez/commit/5ebaeab4164f80539904b9a520d9b7a8307e06e2 as otherwise most HID devices break in funny ways.

**Comment by szaffarano** [comment]
> We probably want to pick [bluez/bluez@5ebaeab](https://github.com/bluez/bluez/commit/5ebaeab4164f80539904b9a520d9b7a8307e06e2) as otherwise most HID devices break in funny ways.

But [the bug](https://github.com/bluez/bluez/issues/880) you pointed out [already exists in 5.75](https://github.com/bluez/bluez/blob/5.75/profiles/input/hog-lib.c#L905) (either in the current stable/unstable version).

Maybe instead of making the release based on this commit, we could wait until `v5.77`, or since the bug is still present in the current nix package, do this release and be aware of `v5.77` to do a new release as soon it's available. As you wish (from my side, I override the package in my local environment because of [this issue](https://github.com/bluez/bluez/issues/815),  then it's fine to me to wait.

Thanks

**Comment by K900** [comment]
We don't need to make a release based on that commit, we can just apply it as patch on top of a stable release, as e.g. SteamOS does (which is actually how I found the patch in the first place).

**Comment by szaffarano** [comment]
> We don't need to make a release based on that commit, we can just apply it as patch on top of a stable release, as e.g. SteamOS does (which is actually how I found the patch in the first place).

Ok, as you wish. I won't be able to spend time on it until at least next week, and I insist that this bug already exists in both the stable and unstable Nix packages

Best.

**Comment by K900** [comment]
FWIW, I'm fine with merging the update as is and picking the patch / next update later, but there's also no real rush as the next staging cycle is at least a few weeks off.

**Comment by AndersonTorres** [comment]
Well, if the bug still persists between versions, the update does not introduce a regression. The patch can be added later in another PR.

**Review by Naxdy — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
I've just pushed the patch to this PR. builds fine for me.

File: pkgs/by-name/bl/bluez/package.nix [hunk 1]
```diff
@@ -18,17 +19,23 @@
...
+    # hog-lib: Fix passing wrong parameters to bt_uhid_get_report_reply
+    (fetchpatch {
+      url = "https://github.com/bluez/bluez/commit/5ebaeab4164f80539904b9a520d9b7a8307e06e2.patch";
+      hash = "sha256-f1A8DmRPfm+zid4XMj1zsfcLZ0WTEax3YPbydKZF9RE=";
+    })
+  ]
     # Disable one failing test with musl libc, also seen by alpine
     # https://github.com/bluez/bluez/issues/726
-    lib.optional (stdenv.hostPlatform.isMusl && stdenv.hostPlatform.isx86_64)
+    ++ lib.optional (stdenv.hostPlatform.isMusl && stdenv.hostPlatform.isx86_64)
```
Comment (AndersonTorres) [inline pkgs/by-name/bl/bluez/package.nix:38]: Is there any problem in patching unconditionally here?

Comment (SuperSandro2000) [reply pkgs/by-name/bl/bluez/package.nix:38]: If the test specifically fails on musl, we shouldn't really disable it everywhere.

**Comment by SuperSandro2000** [comment]
@of org build pkgsMusl.bluez

**Comment by phanirithvij** [comment]
@SuperSandro2000 mistake in the username.

**Comment by phanirithvij** [comment]
@Ofborg build pkgsMusl.bluez

**Pull request merged**